modules/videoio/test/test_audio.cpp

```diff
@@ -267,11 +268,11 @@ TEST_P(Media, audio)
     doTest();
 }
 
-#ifdef _WIN32
 const paramCombination mediaParams[] =
 {
+    paramCombination("test_audio.mp4", 1, 0.15, CV_8UC3, 240, 320, 90, 132299, 30, 30., cv::CAP_GSTREAMER)
```

Comment:
Why do we have doubled "the same" `videoTimestamp`'s on frames 85 and 86 (2866.67 ms)?

```
[ RUN      ] Media.audio/0, where GetParam() = ("test_audio.mp4", 1, 0.15, 16, 240, 320, 90, 132299, 30, 30, GSTREAMER)
[ INFO:0@5.116] global cap_gstreamer.cpp:1378 open OpenCV | GStreamer: /home/alalek/projects/opencv/extra/testdata/highgui/audio/test_audio.mp4
[ INFO:0@5.116] global cap_gstreamer.cpp:1411 open OpenCV | GStreamer: mode - FILE
[ WARN:0@5.607] global cap_gstreamer.cpp:1672 open OpenCV | GStreamer warning: unable to query duration of stream
[ WARN:0@5.610] global cap_gstreamer.cpp:1703 open OpenCV | GStreamer warning: Cannot query video position: status=1, value=0, duration=-1
[ WARN:0@5.630] global cap_gstreamer.cpp:1878 getProperty OpenCV | GStreamer: CAP_PROP_AUDIO_SHIFT_NSEC property is not supported
[ WARN:0@5.630] global cap_gstreamer.cpp:1792 getProperty OpenCV | GStreamer: CAP_PROP_POS_MSEC property result may be unrealiable: https://github.com/opencv/opencv/issues/19025
video0 timestamp: 0  audio0 timestamp: 0 (audio shift nanoseconds: 0 , seconds: 0)
frame=0:  audioFrameSize=1470  videoTimestamp=0 ms
frame=1:  audioFrameSize=1470  videoTimestamp=33.3333 ms
frame=2:  audioFrameSize=1470  videoTimestamp=66.6667 ms
frame=3:  audioFrameSize=1469  videoTimestamp=100 ms
frame=4:  audioFrameSize=1470  videoTimestamp=133.333 ms
frame...
frame=85:  audioFrameSize=0  videoTimestamp=2866.67 ms
frame=86:  audioFrameSize=1470  videoTimestamp=2866.67 ms
frame=87:  audioFrameSize=1470  videoTimestamp=2900 ms
frame=88:  audioFrameSize=1470  videoTimestamp=2933.33 ms
frame=89:  audioFrameSize=1470  videoTimestamp=2966.67 ms
Total audio samples=132299
[       OK ] Media.audio/0 (2821 ms)
```

Reply:
As I wrote, the gst_element_query_position function is not accurate, this is the reason

Reply:
There is some random in output:

```
frame=0:  audioFrameSize=1470  videoTimestamp=0 ms
frame=1:  audioFrameSize=2940  videoTimestamp=33.3333 ms
frame=2:  audioFrameSize=1470  videoTimestamp=100 ms
frame=3:  audioFrameSize=1469  videoTimestamp=133.333 ms
frame=4:  audioFrameSize=1470  videoTimestamp=166.667 ms
frame...
frame=85:  audioFrameSize=1470  videoTimestamp=2866.67 ms
frame=86:  audioFrameSize=1470  videoTimestamp=2900 ms
frame=87:  audioFrameSize=1470  videoTimestamp=2933.33 ms
frame=88:  audioFrameSize=0  videoTimestamp=2966.67 ms
frame=89:  audioFrameSize=1470  videoTimestamp=2966.67 ms
```

```
frame=0:  audioFrameSize=1470  videoTimestamp=0 ms
frame=1:  audioFrameSize=1470  videoTimestamp=33.3333 ms
frame=2:  audioFrameSize=1470  videoTimestamp=66.6667 ms
frame=3:  audioFrameSize=1469  videoTimestamp=100 ms
frame=4:  audioFrameSize=1470  videoTimestamp=133.333 ms
frame...
frame=85:  audioFrameSize=2940  videoTimestamp=2833.33 ms
frame=86:  audioFrameSize=1470  videoTimestamp=2900 ms
frame=87:  audioFrameSize=0  videoTimestamp=2933.33 ms
frame=88:  audioFrameSize=1470  videoTimestamp=2933.33 ms
frame=89:  audioFrameSize=1470  videoTimestamp=2966.67 ms
```

---

Below timestamps for generic video-only scan (ignore crc64):

```
frame=80  frame_size=[320 x 240]  videoTimestamp=2700 ms   crc64=0FF87E8865B65E85
frame=81  frame_size=[320 x 240]  videoTimestamp=2733.33 ms   crc64=5B0016D23525B13E
frame=82  frame_size=[320 x 240]  videoTimestamp=2766.67 ms   crc64=7ACED82BA667C494
frame=83  frame_size=[320 x 240]  videoTimestamp=2800 ms   crc64=5F393F9FF6515BEF
frame=84  frame_size=[320 x 240]  videoTimestamp=2833.33 ms   crc64=1DA3C18F7CC6254F
frame=85  frame_size=[320 x 240]  videoTimestamp=2866.67 ms   crc64=2D2D5461D2C20E89
frame=86  frame_size=[320 x 240]  videoTimestamp=2900 ms   crc64=568E4DC494CEBB11
frame=87  frame_size=[320 x 240]  videoTimestamp=2933.33 ms   crc64=C73EDB4F7625C01D
frame=88  frame_size=[320 x 240]  videoTimestamp=2966.67 ms   crc64=B7BE9F635F4ED293
frame=89  frame_size=[320 x 240]  videoTimestamp=3000 ms   crc64=9349423711753500
```

```.cpp
    VideoCapture capture(findDataFile("audio/test_audio.mp4"), CAP_GSTREAMER);

    Mat frame;
    for (int i = 0; ; i++)
    {
        if (!capture.grab())
            break;
        capture.retrieve(frame);
        std::cout << "frame=" << i << "  frame_size=" << frame.size() << "  videoTimestamp=" << capture.get(CAP_PROP_POS_MSEC) << " ms   " << cv::format("crc64=%016lX", crc64(frame)) << std::endl;
    }
```